src/webgpu/api/validation/createTexture.spec.ts

```diff
@@ -289,19 +289,43 @@ g.test('it_is_invalid_to_submit_a_destroyed_texture_before_and_after_encode')
     }, !_success);
   });
 
-g.test('it_is_invalid_to_have_an_output_attachment_texture_with_non_renderable_format')
-  .params(poptions('format', kAllTextureFormats))
+g.test('texture_usage')
+  .desc(
+    `Test texture usage (single usage or combined usages) for every texture formats and every dimension types`
+  )
+  .subcases(() =>
+    params()
+      .combine(poptions('dimension', kTextureDimensions))
+      .combine(poptions('format', kAllTextureFormats))
+      // If usage0 and usage1 are the same, then the usage being test is a single usage. Otherwise, it is a combined usage.
+      .combine(poptions('usage0', kTextureUsages))
+      .combine(poptions('usage1', kTextureUsages))
+  )
   .fn(async t => {
-    const format: GPUTextureFormat = t.params.format;
+    const { dimension, format, usage0, usage1 } = t.params;
     const info = kAllTextureFormatInfo[format];
 
     await t.selectDeviceOrSkipTestCase(info.extension);
 
-    const descriptor = t.getDescriptor({ width: 1, height: 1, format });
+    const size = dimension === '1d' ? [32, 1, 1] : dimension === '2d' ? [32, 32, 1] : [32, 32, 32];
```

Comment:
nit: the size could even be `[1, 1, 1]` unconditionally.

Reply:
That's right. I just want to make the CTS more like cases in real world. Anyway, I suppose it's not important.